src/hugr/hugrmut.rs

```diff
@@ -115,6 +117,11 @@ pub(crate) trait HugrMut {
     /// In general this invalidates the ports, which may need to be resized to
     /// match the OpType signature.
     fn replace_op(&mut self, node: Node, op: impl Into<OpType>) -> OpType;
+
+    /// Insert another hugr into this one, under a given root node.
+    ///
+    /// Returns the root node of the inserted hugr.
+    fn insert_hugr(&mut self, root: Node, other: &impl AsPortgraph) -> Result<Node, HugrError>;
```

Comment:
I'm not sure making this generic over `impl AsPortgraph` is sound? The method is called `insert_hugr`, it would be helpful if the types stopped me inserting arbitrary portgraphs.

Reply:
We should be able to do `HugrView : sealed::AsPortgraph` and then just require `impl HugrView` here.

Users would still be able to implement `HugrView` on their wrappers if they have `AsRef<Hugr>`